apps/sales/views.py

```diff
@@ -154,4 +156,128 @@ def create(self, request, *args, **kwargs):
         return create_instance(self, request, *args, **kwargs)
 
     def update(self, request, *args, **kwargs):
-        return update_instance(self, request, *args, **kwargs)
\ No newline at end of file
+        return update_instance(self, request, *args, **kwargs)
+
+
+#---------- ONE API - MULTIPLE API CALLS (CRUD OPERATIONS) ---------------
+
+class SaleOrderOneView(generics.GenericAPIView,mi.ListModelMixin,mi.CreateModelMixin,mi.RetrieveModelMixin,mi.UpdateModelMixin,mi.DestroyModelMixin):
+    queryset = SaleOrder.objects.all()
+    serializer_class = SaleOrderSerializer
+
+    def get(self, request, *args, **kwargs):
+            if 'pk' in kwargs:
+                return self.retrieve(request, *args, **kwargs)  # Retrieve a single instance
+            return list_all_objects(self, request, *args, **kwargs)
```

Comment:
A method should have only 1 return statement, use result variable for assigning and return the result varaible

Reply:
discussed in call Thulasi. concluded that it is fine